patches/net/minecraft/world/item/component/ItemContainerContents.java.patch

```diff
@@ -32,6 +32,17 @@
 +            throw new UnsupportedOperationException("Slot " + slot + " not in valid range - [0," + getSlots() + ")");
 +        }
 +    }
++
++    public ItemContainerContents with(int size, int index, ItemStack stack) {
```

Comment:
We either need explicit docs stating that the stack should not be modified after being passed in, or we need to copy the stack when setting it in the list. That being said maybe that isn't the case as `fromItems` seems to copy all the stacks, which is super annoying that they do that now, even though I guess copying is cheaper than it used to be....